docs/content/querying/_index.md

```diff
@@ -90,39 +94,98 @@ method `toSet` is called. The resulting Scala set is converted to a
+are combined via function composition to yield *traversals*. You can
+think of traversals as expressions that describe sets of nodes.
+
+
+
+## Expression Types and Starters
+
+For each node label, a corresponding *expression type* is defined in the
+query language, e.g., `Method` for nodes with the label `METHOD`, and
+`Local` for nodes with the label `LOCAL`. For each label, we also
+define a *starter* that represents all nodes with that label. For
+example,
+
+```bash
+cpg.method
+res1: types.structure.Method = io.shiftleft.semanticcpg.language.types.structure.Method@b889cb6
+```
+returns an expression that yields all methods on evaluation.
+
+You can get a complete list of starters by typing `cpg.`, and
+pressing the Tab key:
+
+```bash
+joern> cpg.
+all             file            local           methodReturn    types
+argument        graph           member          namespace
+call            id              metaData        namespaceBlock
+close           identifier      method          parameter
+comment         literal         methodRef       typeDecl
+
+```
+
+It is generally advisable to use specific starters as opposed to the
+generic `cpg.all`, as for the latter, it is unclear at compile time
+which types of nodes are processed. Using specific starters such as
+`.method` make it possible to use the Scala's type system for
+completion and to detect invalid queries even before evaluation.
+
+## Node IDs
+
+For each node, we store an **ID**. The ID is a long integer that
+uniquely identifies the node.You should not need to ever retrieve
+node IDs within Joern/Ocular scripts, however, they may come in handy
+when passing nodes to external tools - especially if these tools plan
+to run subsequent queries.
+
+As an example, the following query returns the ids of all file nodes.
 
 ```bash
 cpg.file.id.l
 res11: List[AnyRef] = List(34L, 32L, 30L, 28L, 26L, 24L, 22L, 20L, 18L, 16L, 14L, 12L, 10L, 8L, 6L, 4L)
 ```
 
-To select the node with the id `10`, you can use the following query:
+Consider that - upon receiving this result, an external tool would
+like to invoke a follow-up query on the first file node. To achieve
+this, it can issue the query
 
 ```bash
-cpg.id(10).l
+cpg.id(34).l
 res7: List[Nothing] = List(
   File(id -> 10L, name -> "/home/tmp/shiftleft/joern/tarpit-c/tarpitc/buffer_underwrite.c", order -> null)
 )
 ```
 
-Note that the type of the nodes returned by `cpg.id(10)` is unknown at
-compile time, and therefore, the returned type for `cpg.id(10).l`is
-`List[Nothing]`. In practice, you will know the types of nodes you are
-expecting from the previous query. For example, in the outlined
-scenario, we know that case are file nodes. An explicit cast can be
+Note that the type of the nodes returned by `cpg.id(34)` is unknown at
+compile time, and therefore, the returned type for `cpg.id(34).l`is
+`List[Nothing]`. In practice, the external tool will usually know the
```

Comment:
Not related to the content, but a returning a `List[Nothing]` from this seems super strange as it is a bottom type. I would expect `Any` or `AnyRef`.